log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/ThrowableExtendedStackTraceRenderer.java

```diff
@@ -105,12 +107,8 @@ private static Map<String, ClassResourceInfo> createClassResourceInfoByName(
             // But we need the associated `Class<?>` to extract its resource information, i.e., JAR file and version.
             // We are capturing the current stack to find suitable class loaders.
             // We will use this as a bootstrap to go from a class name in a stack trace to a `Class<?>`.
-            final Map<String, ClassResourceInfo> classResourceInfoByName =
-                    StackLocatorUtil.getCurrentStackTrace().stream()
-                            .collect(Collectors.toMap(
-                                    Class::getName,
-                                    clazz -> new ClassResourceInfo(clazz, true),
-                                    (classResourceInfo1, classResourceInfo2) -> classResourceInfo1));
+            final Deque<Class<?>> rootStackTrace = StackLocatorUtil.getCurrentStackTrace();
```

Comment:
I really don't like this deque hack – I know you borrowed it from the earlier version, hence not your fault. It makes things pretty difficult to wrap one's mind around it. I will spend more time on this sometime tomorrow to see if we can do something else...

Reply:
Yeah...it is hard to read...

Reply:
Tried to improve the readability in a7daffed7951a8d995a3390017aaa27e7e0b3df7.